x-pack/plugins/security_solution/public/common/components/filter_group/index.tsx

```diff
@@ -0,0 +1,315 @@
+            }
+          }
+        }
+
+        urlDataApplied.current = true;
+
+        localInitialControls.forEach((control, idx) => {
+          addOptionsListControl(initialInput, {
+            controlId: String(idx),
+            dataViewId,
+            ...control,
+          });
+        });
+      }
+      return initialInput;
+    },
+    [initialControls, dataViewId, timeRange, filters, chainingSystem, query, initialUrlParam]
+  );
+
+  useFilterUpdatesToUrlSync({
+    controlGroupInput: controlGroupInputUpdates,
+  });
+
+  const withContextMenuAction = useCallback(
+    (fn: unknown) => {
+      return () => {
+        if (typeof fn === 'function') {
+          fn();
+        }
+        toggleContextMenu();
+      };
+    },
+    [toggleContextMenu]
+  );
+
+  const resetSelection = useCallback(() => {
+    if (!controlGroupInputUpdates) return;
+
+    const { panels } = controlGroupInputUpdates;
+    Object.values(panels).forEach((control, idx) => {
+      controlGroup?.updateInputForChild(String(idx), {
+        ...control.explicitInput,
+        selectedOptions: initialControls[idx].selectedOptions ?? [],
+        existsSelected: false,
+        exclude: false,
+      });
+    });
+  }, [controlGroupInputUpdates, controlGroup, initialControls]);
+
+  const resetButton = useMemo(
+    () => (
+      <EuiContextMenuItem
+        icon="eraser"
+        onClick={withContextMenuAction(resetSelection)}
+        data-test-subj="filter-group__context--reset"
+      >
+        {`Reset`}
+      </EuiContextMenuItem>
+    ),
+    [withContextMenuAction, resetSelection]
+  );
+
+  const contextMenuItems = useMemo(() => [resetButton], [resetButton]);
+
+  return (
+    <FilterWrapper className="filter-group__wrapper">
+      <EuiFlexGroup alignItems="center" justifyContent="center" gutterSize="s">
+        <EuiFlexItem grow={true} data-test-subj="filter_group__items">
+          <ControlGroupRenderer
+            onLoadComplete={onControlGroupLoadHandler}
+            getInitialInput={setOptions}
+          />
+          {!controlGroup ? (
+            <EuiButton color="text">
+              <EuiLoadingChart data-test-subj="filter-group__loading" />
+            </EuiButton>
+          ) : null}
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiPopover
+            id="filter-group__context-menu"
+            button={
+              <EuiButtonIcon
+                display="empty"
+                size="s"
+                iconType="boxesHorizontal"
+                onClick={toggleContextMenu}
+                data-test-subj="filter-group__context"
+              />
+            }
+            isOpen={isContextMenuVisible}
+            closePopover={toggleContextMenu}
+            panelPaddingSize="none"
+            anchorPosition="downLeft"
+          >
+            <EuiContextMenuPanel items={contextMenuItems} />
+          </EuiPopover>
+        </EuiFlexItem>
+        {/*
+         *<EuiFlexItem grow={false}>
```

Comment:
Figure this is adding functionality for users to manually add filters on the page. Not sure if you want to delete this, or add a comment explaining why it's currently commented out?

Reply:
@michaelolo24 

Yes, that was the initial intention so that the component is composable and easily understandable but current controls plugin do no support this way to defining components so I had to change the strategy.

I will remove it. Thanks.